### PR TITLE
Add slug input/edit field in Event form.

### PR DESCRIPTION
**Pull request by EslamHiko**
this is a PR to solve: https://github.com/opencollective/opencollective/issues/1231

Create Event : 

crea

Edit Event:

edit

Submitting API PRs in a moment

### CODE REVIEW

File: components/EditEventForm.js [hunk 1]
```diff
@@ -80,15 +80,30 @@ class EditEventForm extends React.Component {
...
       }
     }
+    let slugChanged = this.state.slugChanged;
+    if (fieldname === 'slug') {
+      slugChanged = true;
+    }
+    if (!slugChanged && fieldname === 'name') {
+      event['slug'] = event['name'].length
+        ? `${slugify(event['name'])}-${this.state.event.parentCollective.id}ev`.toLowerCase()
+        : `{eventName}-${this.state.event.parentCollective.id}ev`.toLowerCase();
```
Comment (Betree) [inline components/EditEventForm.js]: Not sure about this line, having `{eventName}-4222ev` displayed as a slug will not look nice. I'd just remove the first part.

```suggestion
        : `-${this.state.event.parentCollective.id}ev`;
```

File: components/EditEventForm.js [hunk 2]
```diff
@@ -80,15 +80,30 @@ class EditEventForm extends React.Component {
         event['endsAt'] = value;
       }
     }
+    let slugChanged = this.state.slugChanged;
+    if (fieldname === 'slug') {
+      slugChanged = true;
+    }
+    if (!slugChanged && fieldname === 'name') {
+      event['slug'] = event['name'].length
+        ? `${slugify(event['name'])}-${this.state.event.parentCollective.id}ev`.toLowerCase()
```
Comment (Betree) [inline components/EditEventForm.js]: `slugify` already set string to lowercase

```suggestion
        ? `${slugify(event['name'])}-${this.state.event.parentCollective.id}ev`
```

**Comment by Betree** [comment]
To make CI happy, you should commit your `package-lock.json`

File: components/EditEventForm.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { defineMessages, injectIntl } from 'react-intl';
-
```
Comment (znarf) [inline components/EditEventForm.js]: We already have a `slugify` package in our `package.json`, let's use this one?

**Comment by xdamman** [comment]
Not a fan of this. If we let users define the slug of an event, this will occupy prime real estate in our slug space. Eventbrite and medium and others also append a random number at the end to avoid this problem.

File: components/EditEventForm.js (hunk 3 above)
Comment (Betree) [reply components/EditEventForm.js]: We use `limax` (which has better support for special characters, https://github.com/opencollective/opencollective-api/pull/1480#issuecomment-429791678) on the API, wouldn't be a bad idea to make the switch for the Frontend.

But I agree that this should be part of a separate PR dedicated to that, we should use `slugify` for now.

**Comment by Betree** [comment]
@xdamman has a fair point, it would be annoying to have events taking slugs to collectives. Maybe we should enforce `-${this.state.event.parentCollective.id}ev` at the end, even when using a custom slug.

**Comment by EslamHiko** [comment]
@Betree so we will only show the slug without `-${this.state.event.parentCollective.id}ev` in the edit form and append it in the backend

**Comment by EslamHiko** [comment]
@Betree I think now we won't need the API modification
Now it'll always append `-{collectiveId}ev` to the slug.

**Comment by znarf** [comment]
`limax` might have a huge footprint in bundle size, to be checked

**Comment by znarf** [comment]
@Betree @xdamman I don't agree with this notion of "prime real estate in our slug namespace". I find that adding "-{collectiveId}ev" to all events is pretty ugly, this issue was an opportunity to revisit this.

**Comment by EslamHiko** [comment]
My last commit is renaming the slug of events. check the gif

versions

**Comment by Betree** [comment]
> @Betree @xdamman I don't agree with this notion of "prime real estate in our slug namespace". I find that adding "-{collectiveId}ev" to all events is pretty ugly, this issue was an opportunity to revisit this.

Because event's full URLs are `/{collective}/events/{event_slug}`, I think that most users would expect `event_slug` to be unique to my collective, not to all collectives/events. 
Example: I'd like to be able to create an event like `/paris-js/events/react-meetup`, and shouldn't be prevented to do so just because there's already another event/collective with this name.

If we make an analogy with Github, that would be similar to preventing the creation of a repo under `https://github.com/betree/site-perso` because there's already a repo like `https://github.com/jean/site-perso`.

Events are collectives, so it's complex for us to solve this properly. I agree that adding `-{collectiveId}ev` is not the cleanest, but I think that the alternative of preventing you to use slugs because there are already other collectives/events with this name is worse in terms of user experience.

**Comment by znarf** [comment]
We decided to not do that and solve the problem differently.

Still, thank you for your work on that @EslamHiko, appreciated !

**Pull request closed without merging**